server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java

```diff
@@ -65,7 +65,14 @@ public ActiveRuleIndex(EsClient client) {
     return countByField(FIELD_ACTIVE_RULE_PROFILE_KEY,
       QueryBuilders.hasParentQuery(TYPE_RULE,
         QueryBuilders.boolQuery().mustNot(
-          QueryBuilders.termQuery(FIELD_RULE_STATUS, "REMOVED"))));
+          QueryBuilders.termQuery(FIELD_RULE_STATUS, RuleStatus.REMOVED.name()))));
+  }
+
+  public Map<String, Long> countAllDeprecatedByQualityProfileKey() {
+    return countByField(FIELD_ACTIVE_RULE_PROFILE_KEY,
```

Comment:
Cool ! I thought you would not be able to use countByField(), but it good to know that it could have been used to cover this use case.